presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -480,7 +496,7 @@ public boolean mkdirs(Path f, FsPermission permission)
         return true;
     }
 
-    private Iterator<LocatedFileStatus> listPrefix(Path path)
+    private Iterator<LocatedFileStatus> listPrefix(Path path, OptionalInt initialMaxKeys, boolean recursive, boolean filterDirectories)
```

Comment:
for filterDirectories, shall we use returnFilesOnly? for easy understanding

Reply:
I've converted this to an enum in the latest revision because the two boolean mode flags could represent a potentially invalid request to list recursively including directories, which isn't a native S3 operation.